### ISSUE TITLE
Support for Attachments

### ISSUE THREAD

**Issue by dullage** [dullage/flatnotes#134]
This functionality would be similar to the image support released in v3.4.0 but would allow users to upload any file type after which a download URL would be added to the editor.

**Comment by dobrKot** [comment]
Oh, I sign the petition for the implementation as well.

**Comment by monkeyx-net** [comment]
Plus one from me.

I have worked out a hacky work around though :)

```
<a  href="attachments/test.pdf" target="_blank">PDF Link</a>
```

I add the pdf as image then change the link to above and it open a new tab :)

```
[PDF LINK](attachments/test.pdf){:target="_blank"}
```

The above works if I right click and open in new tab. I include the the reference {} as it does not work. But some markdown parsers do support this

**Comment by dobrKot** [comment]
> Plus one from me.
> 
> I have worked out a hacky work around though :)
> 
> ```
> <a  href="attachments/test.pdf" target="_blank">PDF Link</a>
> ```
> 
> I add the pdf as image then change the link to above and it open a new tab :)
> 
> ```
> [PDF LINK](attachments/test.pdf){:target="_blank"}
> ```
> 
> The above works if I right click and open in new tab. I include the the reference {} as it does not work. But some markdown parsers do support this

Yeah, I know. I went through code and the workflow is basically supporting any type of files already, with minor changes, and with renaming of "uploadImage()" to "uploadAttachment()". There till will be some questions like: setting up the maximum file size, adding some basic progress bar and some mechanism to delete attachment together with note(it's not that crucial with pics, but will bloat the hdd much faster with bigger files).

Right now I am thinking betwen choosing other app like Memos(which has cons as well), or proceed with Flatnotes and some kind of file system manager, where all files will be stored, like FileGator, etc.

**Comment by dd0ke** [comment]
Oops, that was me on that duplicate in #262! I'd done some Google and Perplexity homework, but I did not browse the open issues. Sorry.

So if anyone is in doubt, +1 for libre uploads!

**Comment by jacount** [comment]
Thank you for a great project @dullage ! Plus one from me to be able to upload pdfs to a folder
